### PR TITLE
Adding Fibbonacci and Golden_Search_Method which measure the Itteration more precisely.

### PR DESCRIPTION
**Pull request by Niteshshah123**
These Two Method (i.e Fibonnacci and Golden_Search method) Optimize the equation more Preciously. File named Fibbonacci optimize the equation based on number of itteration. whereas File named Fibbonacci2 optimize the equation based on the Percentage of accuracy. like, if the user want the result more accurate value (such as: the result should lie between the interval of 1% of actual minimum value) without caring the number of itteration then fibbonacci2 is best otherwise if user want to solve it using less itteration without caring much on accuracy, then Fibonacci is best.

### CODE REVIEW

**Review by ShudarsanRegmi — CHANGES_REQUESTED** [review]
Please make following changes to you PR:

- Don't write two separate files for the same program with different inputs. Instead you can ask user which one of the input they want to make
- Please keep track of your iteration tables. You're just printing the values from the loop. Instead you should iniitalize a list named `table` and a list `row` in each iteration so that we can easily pass the information to the UI for rendering.
- If you're writing the program  which has already been done, please prefix the name of the algorithm with the first two letter of your first and last name, respectively. So that we can compare all the implementations and merge the best implementation.

**Comment by Niteshshah123** [comment]
Now, I have added table to show output. It will help to improve user UI Experience.

**Comment by ShudarsanRegmi** [comment]
Please create only one file for the Fibonacci search algorithm.

**Comment by ShudarsanRegmi** [comment]
Please add one more column for 'L/R' for knowing the next chosen interval. 
image
Like above

File: utils/NS_fibbonacci2.py
Comment (ShudarsanRegmi) [inline utils/NS_fibbonacci2.py:1]: Please provide Column for 'L/R'

File: utils/NS_fibbonacci.py
Comment (ShudarsanRegmi) [inline utils/NS_fibbonacci.py:1]: Please add column for 'L/R'

File: utils/NS_fibbonacci2.py
Comment (ShudarsanRegmi) [inline utils/NS_fibbonacci2.py:1]: Please find out why the table is getting deviated for later iterations. 
The output table for your first implementation is: 
image
For this implementation, your table output is: 
image

The later rows in the second table seems to be getting deviated.
Also, why are there only 13 iterations for getting 10% accuracy on this one? for your previous implementation it was 15 iterations for the same 10%.

The optimum point and value both seems to be wrong.

image

Please mention the  reason for choosing the new intervals?
Update the important values(such as value of a, b, functional values, etc) in each of your drop of a or b.

At last, also mention the column for 'L/R' operation for this one as well.

**Review by ShudarsanRegmi — CHANGES_REQUESTED** [review]

**Comment by Niteshshah123** [comment]
yes,  there only 13 iterations for getting 10% accuracy on this one? for
your previous implementation it was 15 iterations for the same 10%. because
both method is different(I have also mention it before). user can use any
one method based on the iteration or precision. See! the program with 15
itteration seems to be more accurate then 13 itteration. and if you want to
implement only one function, then you can choose any one of them based on
your preference and the precision value.
and now for tabulating the output, i have already shown the output as
table. But if you want to modify the table, then explain a bit about the
format(means, where i am lagging and what new should be adding in that
table).

and this image
[image: image.png]
is showing which one i am elementing and which one i am preserving. like
elementing 'b' means elementing 'R' and preserving 'a' means preserving
'L'.  I have used "a" and "b" instead of "l" and "R". if you want to
modify. Let me know about that.

and you also suggest to mention  ''L/R' operation at last, but i am not
able to understand. means what kind of operation you are talking about. can
you send me the image for the reference of other method to get, what
exactly you are trying to suggest.

File: utils/tempCodeRunnerFile.py
Comment (ShudarsanRegmi) [inline utils/tempCodeRunnerFile.py:1]: Don't push file like this. 
If these like files are generating, simply put them in .gitignore

**Comment by Niteshshah123** [comment]
---------- Forwarded message ---------
From: Cir Amritapuri ***@***.***>
Date: Mon, 30 Mar 2026, 14:54
Subject: JPMorganChase | Assessment Instructions - Full-Time 2027POB
To: ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.***
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.***
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.***
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.***
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, Aditya
Kothapalli ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.***
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.***
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.***
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.***
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.***
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, Akhilesh Mohan ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.***
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.***
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>, ***@***.***
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** <
***@***.***>, ***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>,
***@***.*** ***@***.***>


Dear Students,
Please see the update below from *JPMorganChase regarding the online
assessment for B.Tech 2027 POB.*
Check the email below for details on the assessment time, instructions, and
test format. *The test links will be sent directly to the registered email
IDs of shortlisted students.*
All students are requested to complete the test within the given timeline.


Thanks & Regards,

*Training & Placement*
*Corporate & Industry Relations*
*Amrita Vishwa Vidyapeetham*
*Amritapuri Campus*



------------------------------

*Subject:* UPDATE: JPMorganChase | Assessment Instructions - Full-Time 2027






Dear All,



Greetings from JPMorganChase!



Thank you for your continued support and partnership with JPMorganChase. We
are pleased to provide important instructions for students shortlisted for
the Full-Time Software Engineer Program (SEP) 2027.



ASSESSMENT PROCESS OVERVIEW



The SEP 2027 assessment process consists of three stages:

   1. HackerRank Online Coding Assessment
   2. HireVue Video Interview (for candidates who clear HackerRank
   assessment)
   3. Code for Good Event



*STAGE 1: HACKERRANK ONLINE CODING ASSESSMENT*



*Online Test Start Date & Time : **31st March 12:00 AM IST*

*Online Test End Date & Time: **1st April 11:59 PM IST*



*Access & Registration**:* Unique test links will be sent directly to
shortlisted students’ registered email addresses. When starting the test,
students must enter the same personal details (name, email ID, etc.) that
were used during their application on the JPMorganChase careers page.



*Test Format**:* The assessment will be conducted on the HackerRank
platform and consists of 2 questions (50 marks each). Students will have 60
minutes to complete the test with no negative marking. The test must be
completed in a single sitting within the specified timeline, and access
will not be available outside these dates.



*For technical support and FAQs visit:*

*https://candidatesupport.hackerrank.com/hc/en-us/sections/4402845071635-Test?secureweb=Teams*
<https://candidatesupport.hackerrank.com/hc/en-us/sections/4402845071635-Test?secureweb=Teams>



*STAGE 2: HIREVUE VIDEO INTERVIEW*



*Eligibility & Timeline*: Candidates who successfully clear the HackerRank
assessment will receive a unique link for the video interview via HireVue.
Students can complete the interview at their convenience anytime before
April 4 at 11:59 PM IST.



*Interview Format:* The video interview consists of 2 questions and should
take approximately 30 minutes to complete. The interview must be completed
in a single sitting.



*NOTE*: Both assessments are proctored. Students are strictly prohibited
from switching tabs, copying from others or LLMs, searching for solutions
online or using any external sources. Violation of these guidelines will
result in immediate disqualification. We encourage students to attempt all
questions to maximize their score.



Students who do not qualify at any stage will be notified directly via
their registered email address. The list of students selected for the Code
for Good event will be shared by the 2nd week of May.



Should you have any queries, please feel free to reach out to me.



We wish all shortlisted candidates the very best in the process!

















© 2025 JPMorgan Chase & Co. All rights reserved.

JPMorgan Chase is an Equal Opportunity Employer, including
Disability/Veterans









[image: Amrita University]
Disclaimer: The information transmitted in this email, including
attachments, is intended only for the person(s) or entity to which it is
addressed and may contain confidential and/or privileged material. Any
review, retransmission, dissemination or other use of, or taking of any
action in reliance upon this information by persons or entities other than
the intended recipient is prohibited. Any views expressed in any message
are those of the individual sender and may not necessarily reflect the
views of Amrita Vishwa Vidyapeetham. If you received this in error, please
contact the sender and destroy any copies of this information.

**Pull request merged**